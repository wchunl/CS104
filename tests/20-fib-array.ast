ROOT "" (0.0.0)
|   TOK_FUNCTION "" (6.22.1)
|   |   TOK_TYPE_ID "" (6.22.1)
|   |   |   TOK_VOID "void" (6.22.1)
|   |   |   TOK_IDENT "fail" (6.22.6)
|   |   TOK_PARAM "(" (6.22.11)
|   |   |   TOK_TYPE_ID "" (6.22.12)
|   |   |   |   TOK_STRING "string" (6.22.12)
|   |   |   |   TOK_IDENT "expr" (6.22.19)
|   |   |   TOK_TYPE_ID "" (6.22.25)
|   |   |   |   TOK_STRING "string" (6.22.25)
|   |   |   |   TOK_IDENT "file" (6.22.32)
|   |   |   TOK_TYPE_ID "" (6.22.38)
|   |   |   |   TOK_INT "int" (6.22.38)
|   |   |   |   TOK_IDENT "line" (6.22.42)
|   TOK_FUNCTION "" (6.24.1)
|   |   TOK_TYPE_ID "" (6.24.1)
|   |   |   TOK_VOID "void" (6.24.1)
|   |   |   TOK_IDENT "putchr" (6.24.6)
|   |   TOK_PARAM "(" (6.24.13)
|   |   |   TOK_TYPE_ID "" (6.24.14)
|   |   |   |   TOK_INT "int" (6.24.14)
|   |   |   |   TOK_IDENT "chr" (6.24.18)
|   TOK_FUNCTION "" (6.25.1)
|   |   TOK_TYPE_ID "" (6.25.1)
|   |   |   TOK_VOID "void" (6.25.1)
|   |   |   TOK_IDENT "putint" (6.25.6)
|   |   TOK_PARAM "(" (6.25.13)
|   |   |   TOK_TYPE_ID "" (6.25.14)
|   |   |   |   TOK_INT "int" (6.25.14)
|   |   |   |   TOK_IDENT "num" (6.25.18)
|   TOK_FUNCTION "" (6.26.1)
|   |   TOK_TYPE_ID "" (6.26.1)
|   |   |   TOK_VOID "void" (6.26.1)
|   |   |   TOK_IDENT "putstr" (6.26.6)
|   |   TOK_PARAM "(" (6.26.13)
|   |   |   TOK_TYPE_ID "" (6.26.14)
|   |   |   |   TOK_STRING "string" (6.26.14)
|   |   |   |   TOK_IDENT "str" (6.26.21)
|   TOK_FUNCTION "" (6.28.1)
|   |   TOK_TYPE_ID "" (6.28.1)
|   |   |   TOK_INT "int" (6.28.1)
|   |   |   TOK_IDENT "getchr" (6.28.5)
|   |   TOK_PARAM "(" (6.28.11)
|   TOK_FUNCTION "" (6.29.1)
|   |   TOK_TYPE_ID "" (6.29.1)
|   |   |   TOK_STRING "string" (6.29.1)
|   |   |   TOK_IDENT "getstr" (6.29.8)
|   |   TOK_PARAM "(" (6.29.14)
|   TOK_FUNCTION "" (6.30.1)
|   |   TOK_TYPE_ID "" (6.30.1)
|   |   |   TOK_STRING "string" (6.30.1)
|   |   |   TOK_IDENT "getln" (6.30.8)
|   |   TOK_PARAM "(" (6.30.13)
|   TOK_FUNCTION "" (7.10.1)
|   |   TOK_TYPE_ID "" (7.10.1)
|   |   |   TOK_INT "int" (7.10.1)
|   |   |   TOK_IDENT "main" (7.10.5)
|   |   TOK_PARAM "(" (7.10.9)
|   |   TOK_BLOCK "{" (7.10.12)
|   |   |   TOK_VARDECL "=" (7.11.25)
|   |   |   |   TOK_ARRAY "array" (7.11.4)
|   |   |   |   |   TOK_TYPE_ID "int" (7.11.10)
|   |   |   |   TOK_IDENT "fibonacci" (7.11.15)
|   |   |   |   TOK_ALLOC "alloc" (7.11.27)
|   |   |   |   |   TOK_ARRAY "array" (7.11.33)
|   |   |   |   |   TOK_TYPE_ID "int" (7.11.39)
|   |   |   |   |   TOK_INTCON "30" (7.11.46)
|   |   |   '=' "=" (7.12.17)
|   |   |   |   TOK_INDEX "[" (7.12.13)
|   |   |   |   |   TOK_IDENT "fibonacci" (7.12.4)
|   |   |   |   |   TOK_INTCON "0" (7.12.14)
|   |   |   |   TOK_INTCON "0" (7.12.19)
|   |   |   '=' "=" (7.13.17)
|   |   |   |   TOK_INDEX "[" (7.13.13)
|   |   |   |   |   TOK_IDENT "fibonacci" (7.13.4)
|   |   |   |   |   TOK_INTCON "1" (7.13.14)
|   |   |   |   TOK_INTCON "1" (7.13.19)
|   |   |   TOK_VARDECL "=" (7.14.14)
|   |   |   |   TOK_INT "int" (7.14.4)
|   |   |   |   TOK_IDENT "index" (7.14.8)
|   |   |   |   TOK_INTCON "2" (7.14.16)
|   |   |   TOK_WHILE "while" (7.15.4)
|   |   |   |   '<' "<" (7.15.17)
|   |   |   |   |   TOK_IDENT "index" (7.15.11)
|   |   |   |   |   TOK_INTCON "30" (7.15.19)
|   |   |   |   TOK_BLOCK "{" (7.15.23)
|   |   |   |   |   '=' "=" (7.16.24)
|   |   |   |   |   |   TOK_INDEX "[" (7.16.16)
|   |   |   |   |   |   |   TOK_IDENT "fibonacci" (7.16.7)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.16.17)
|   |   |   |   |   |   '+' "+" (7.16.47)
|   |   |   |   |   |   |   TOK_INDEX "[" (7.16.35)
|   |   |   |   |   |   |   |   TOK_IDENT "fibonacci" (7.16.26)
|   |   |   |   |   |   |   |   '-' "-" (7.16.42)
|   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.16.36)
|   |   |   |   |   |   |   |   |   TOK_INTCON "1" (7.16.44)
|   |   |   |   |   |   |   TOK_INDEX "[" (7.16.58)
|   |   |   |   |   |   |   |   TOK_IDENT "fibonacci" (7.16.49)
|   |   |   |   |   |   |   |   '-' "-" (7.16.65)
|   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.16.59)
|   |   |   |   |   |   |   |   |   TOK_INTCON "2" (7.16.67)
|   |   |   |   |   '=' "=" (7.17.13)
|   |   |   |   |   |   TOK_IDENT "index" (7.17.7)
|   |   |   |   |   |   '+' "+" (7.17.21)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.17.15)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.17.23)
|   |   |   '=' "=" (7.19.10)
|   |   |   |   TOK_IDENT "index" (7.19.4)
|   |   |   |   TOK_INTCON "0" (7.19.12)
|   |   |   TOK_CALL "(" (7.20.11)
|   |   |   |   TOK_IDENT "putstr" (7.20.4)
|   |   |   |   TOK_STRINGCON ""Numeri di figlio Bonacci\n"" (7.20.12)
|   |   |   TOK_WHILE "while" (7.21.4)
|   |   |   |   '<' "<" (7.21.17)
|   |   |   |   |   TOK_IDENT "index" (7.21.11)
|   |   |   |   |   TOK_INTCON "30" (7.21.19)
|   |   |   |   TOK_BLOCK "{" (7.21.23)
|   |   |   |   |   TOK_CALL "(" (7.22.14)
|   |   |   |   |   |   TOK_IDENT "putstr" (7.22.7)
|   |   |   |   |   |   TOK_STRINGCON ""fibonacci["" (7.22.15)
|   |   |   |   |   TOK_CALL "(" (7.23.14)
|   |   |   |   |   |   TOK_IDENT "putint" (7.23.7)
|   |   |   |   |   |   TOK_IDENT "index" (7.23.15)
|   |   |   |   |   TOK_CALL "(" (7.24.14)
|   |   |   |   |   |   TOK_IDENT "putstr" (7.24.7)
|   |   |   |   |   |   TOK_STRINGCON ""] = "" (7.24.15)
|   |   |   |   |   TOK_CALL "(" (7.25.14)
|   |   |   |   |   |   TOK_IDENT "putint" (7.25.7)
|   |   |   |   |   |   TOK_INDEX "[" (7.25.24)
|   |   |   |   |   |   |   TOK_IDENT "fibonacci" (7.25.15)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.25.25)
|   |   |   |   |   TOK_CALL "(" (7.26.14)
|   |   |   |   |   |   TOK_IDENT "putchr" (7.26.7)
|   |   |   |   |   |   TOK_CHARCON "'\n'" (7.26.15)
|   |   |   |   |   '=' "=" (7.27.13)
|   |   |   |   |   |   TOK_IDENT "index" (7.27.7)
|   |   |   |   |   |   '+' "+" (7.27.21)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.27.15)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.27.23)
|   |   |   TOK_RETURN "return" (7.29.4)
|   |   |   |   TOK_INTCON "0" (7.29.11)
