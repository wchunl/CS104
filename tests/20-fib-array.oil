fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
main:     .function int
          .local ptr fibonacci
          .local int index
          malloc(4 * sizeof int)
          fibonacci = $t0:i
          fibonacci[0 * :p] = 0
          fibonacci[1 * :p] = 1
          index = 2
.wh0:     $t1:i = index < 30
          goto .od0 if not $t1:i
.do0:     $t2:p = fibonacci[- * :p]
          $t3:i = $t2:i + [
          fibonacci[index * :p] = $t3:i
          $t4:i = index + 1
          index = $t4:i
.od0:     index = 0
          putstr("Numeri di figlio Bonacci\n")
.wh1:     $t5:i = index < 30
          goto .od1 if not $t5:i
.do1:     putstr("fibonacci[")
          putint(index)
          putstr("] = ")
          $t6:p = fibonacci[index * :p]
          putint($t6:i)
          putchr('\n')
          $t7:i = index + 1
          index = $t7:i
.od1:     return 0
          return 
          .end
