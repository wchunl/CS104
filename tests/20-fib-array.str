string_set[   0]:    9213716583603343677 0x17bf088->""fibonacci[""
                    4706206694200543916 0x17bbc08->","
                   16579916199511856337 0x17bd098->"getln"
string_set[   1]:    2297668033614959926 0x17bdd88->"0"
string_set[   3]:   11868120863286193964 0x17bd2b8->"main"
                    9611428142580219387 0x17b7608->"fail"
string_set[   4]:      21606623943142546 0x17bd508->"{"
                    4017852018407011111 0x17bbcf8->"file"
string_set[   6]:   13686721134187162920 0x17beb58->"}"
string_set[   8]:    1370730858159036685 0x17be8e8->"+"
                    9386170080689500517 0x17bbad8->"expr"
string_set[   9]:    3560073992529684641 0x17bf438->""] = ""
                   10319632053171904911 0x17be2e8->"while"
                   14526079891305879776 0x17bd7f8->">"
string_set[  12]:   15816506932648033945 0x17bb968->"("
string_set[  17]:   18413295556186398326 0x17bc0f8->"putint"
string_set[  18]:    7211130878751977441 0x17bc228->"putchr"
                   12679825168387489523 0x17bc708->"num"
string_set[  21]:   13581035385482130682 0x17bdd38->"["
string_set[  23]:   12051031859636400188 0x17bddd8->"]"
                    8690754626977038595 0x17bdc28->"30"
                   11166701219533395393 0x17bce58->"getstr"
                   12868445110721718657 0x17bbd18->")"
string_set[  28]:   14635384904416252381 0x17bfcb8->"return"
                   16378307159501682956 0x17bd868->"fibonacci"
string_set[  29]:   12291392672316406109 0x17bbea8->"line"
string_set[  31]:    6192799629375248351 0x17bf5b8->"'\n'"
                    6142509188972423790 0x17b75e8->""
                   13348923631108115385 0x17bc4e8->"putstr"
string_set[  32]:   11597697714117577063 0x17bd818->"="
string_set[  33]:    8290819294854170237 0x17bd958->"alloc"
                    1583576014053361350 0x17bc888->"str"
string_set[  34]:    7372855250431704371 0x17be4e8->"-"
string_set[  38]:   14002868460244944809 0x17bee58->""Numeri di figlio Bonacci\n""
                   12597418120160561823 0x17bbb48->"int"
string_set[  39]:    7296690267450279874 0x17bc368->"chr"
string_set[  40]:   11911796782241914524 0x17b7218->"void"
string_set[  42]:    4698864730501174006 0x17bddf8->"index"
                    5344662657899890615 0x17bbec8->";"
string_set[  43]:    4551451650890805270 0x17be248->"2"
                    6814028903552918164 0x17bc038->"array"
string_set[  44]:   13497578857207247412 0x17bba08->"string"
string_set[  45]:   10159970873491820195 0x17bdce8->"1"
                    1396094175042499165 0x17bc058->"<"
string_set[  46]:    9201800968376702500 0x17bcc38->"getchr"
load_factor = 0.957
bucket_count = 47
max_bucket_size = 4
