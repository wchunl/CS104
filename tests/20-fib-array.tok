#  1 "20-fib-array.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "20-fib-array.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  7 "20-fib-array.oc"
   7 10.00001 262 TOK_INT       int
   7 10.00005 282 TOK_IDENT     main
   7 10.00009  40 '('           (
   7 10.00010  41 ')'           )
   7 10.00012 123 '{'           {
   7 11.00004 271 TOK_ARRAY     array
   7 11.00009  60 '<'           <
   7 11.00010 262 TOK_INT       int
   7 11.00013  62 '>'           >
   7 11.00015 282 TOK_IDENT     fibonacci
   7 11.00025  61 '='           =
   7 11.00027 273 TOK_ALLOC     alloc
   7 11.00032  60 '<'           <
   7 11.00033 271 TOK_ARRAY     array
   7 11.00038  60 '<'           <
   7 11.00039 262 TOK_INT       int
   7 11.00042  62 '>'           >
   7 11.00043  62 '>'           >
   7 11.00045  40 '('           (
   7 11.00046 283 TOK_INTCON    30
   7 11.00048  41 ')'           )
   7 11.00049  59 ';'           ;
   7 12.00004 282 TOK_IDENT     fibonacci
   7 12.00013  91 '['           [
   7 12.00014 283 TOK_INTCON    0
   7 12.00015  93 ']'           ]
   7 12.00017  61 '='           =
   7 12.00019 283 TOK_INTCON    0
   7 12.00020  59 ';'           ;
   7 13.00004 282 TOK_IDENT     fibonacci
   7 13.00013  91 '['           [
   7 13.00014 283 TOK_INTCON    1
   7 13.00015  93 ']'           ]
   7 13.00017  61 '='           =
   7 13.00019 283 TOK_INTCON    1
   7 13.00020  59 ';'           ;
   7 14.00004 262 TOK_INT       int
   7 14.00008 282 TOK_IDENT     index
   7 14.00014  61 '='           =
   7 14.00016 283 TOK_INTCON    2
   7 14.00017  59 ';'           ;
   7 15.00004 267 TOK_WHILE     while
   7 15.00010  40 '('           (
   7 15.00011 282 TOK_IDENT     index
   7 15.00017  60 '<'           <
   7 15.00019 283 TOK_INTCON    30
   7 15.00021  41 ')'           )
   7 15.00023 123 '{'           {
   7 16.00007 282 TOK_IDENT     fibonacci
   7 16.00016  91 '['           [
   7 16.00017 282 TOK_IDENT     index
   7 16.00022  93 ']'           ]
   7 16.00024  61 '='           =
   7 16.00026 282 TOK_IDENT     fibonacci
   7 16.00035  91 '['           [
   7 16.00036 282 TOK_IDENT     index
   7 16.00042  45 '-'           -
   7 16.00044 283 TOK_INTCON    1
   7 16.00045  93 ']'           ]
   7 16.00047  43 '+'           +
   7 16.00049 282 TOK_IDENT     fibonacci
   7 16.00058  91 '['           [
   7 16.00059 282 TOK_IDENT     index
   7 16.00065  45 '-'           -
   7 16.00067 283 TOK_INTCON    2
   7 16.00068  93 ']'           ]
   7 16.00069  59 ';'           ;
   7 17.00007 282 TOK_IDENT     index
   7 17.00013  61 '='           =
   7 17.00015 282 TOK_IDENT     index
   7 17.00021  43 '+'           +
   7 17.00023 283 TOK_INTCON    1
   7 17.00024  59 ';'           ;
   7 18.00004 125 '}'           }
   7 19.00004 282 TOK_IDENT     index
   7 19.00010  61 '='           =
   7 19.00012 283 TOK_INTCON    0
   7 19.00013  59 ';'           ;
   7 20.00004 282 TOK_IDENT     putstr
   7 20.00011  40 '('           (
   7 20.00012 285 TOK_STRINGCON "Numeri di figlio Bonacci\n"
   7 20.00040  41 ')'           )
   7 20.00041  59 ';'           ;
   7 21.00004 267 TOK_WHILE     while
   7 21.00010  40 '('           (
   7 21.00011 282 TOK_IDENT     index
   7 21.00017  60 '<'           <
   7 21.00019 283 TOK_INTCON    30
   7 21.00021  41 ')'           )
   7 21.00023 123 '{'           {
   7 22.00007 282 TOK_IDENT     putstr
   7 22.00014  40 '('           (
   7 22.00015 285 TOK_STRINGCON "fibonacci["
   7 22.00027  41 ')'           )
   7 22.00028  59 ';'           ;
   7 23.00007 282 TOK_IDENT     putint
   7 23.00014  40 '('           (
   7 23.00015 282 TOK_IDENT     index
   7 23.00020  41 ')'           )
   7 23.00021  59 ';'           ;
   7 24.00007 282 TOK_IDENT     putstr
   7 24.00014  40 '('           (
   7 24.00015 285 TOK_STRINGCON "] = "
   7 24.00021  41 ')'           )
   7 24.00022  59 ';'           ;
   7 25.00007 282 TOK_IDENT     putint
   7 25.00014  40 '('           (
   7 25.00015 282 TOK_IDENT     fibonacci
   7 25.00024  91 '['           [
   7 25.00025 282 TOK_IDENT     index
   7 25.00030  93 ']'           ]
   7 25.00031  41 ')'           )
   7 25.00032  59 ';'           ;
   7 26.00007 282 TOK_IDENT     putchr
   7 26.00014  40 '('           (
   7 26.00015 284 TOK_CHARCON   '\n'
   7 26.00019  41 ')'           )
   7 26.00020  59 ';'           ;
   7 27.00007 282 TOK_IDENT     index
   7 27.00013  61 '='           =
   7 27.00015 282 TOK_IDENT     index
   7 27.00021  43 '+'           +
   7 27.00023 283 TOK_INTCON    1
   7 27.00024  59 ';'           ;
   7 28.00004 125 '}'           }
   7 29.00004 268 TOK_RETURN    return
   7 29.00011 283 TOK_INTCON    0
   7 29.00012  59 ';'           ;
   7 30.00001 125 '}'           }
