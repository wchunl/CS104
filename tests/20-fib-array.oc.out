; 0.  1: # 1 "20-fib-array.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "20-fib-array.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: 
; 4.  4: 
; 4.  5: 
; 4.  6: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 7 "20-fib-array.oc" 2
; 7.  7: 
; 7.  8: 
; 7.  9: 
; 7. 10: int main() {
; 7. 11:    array<int> fibonacci = alloc<array<int>> (30);
; 7. 12:    fibonacci[0] = 0;
; 7. 13:    fibonacci[1] = 1;
; 7. 14:    int index = 2;
; 7. 15:    while (index < 30) {
; 7. 16:       fibonacci[index] = fibonacci[index - 1] + fibonacci[index - 2];
; 7. 17:       index = index + 1;
; 7. 18:    }
; 7. 19:    index = 0;
; 7. 20:    putstr ("Numeri di figlio Bonacci\n");
; 7. 21:    while (index < 30) {
; 7. 22:       putstr ("fibonacci[");
; 7. 23:       putint (index);
; 7. 24:       putstr ("] = ");
; 7. 25:       putint (fibonacci[index]);
; 7. 26:       putchr ('\n');
; 7. 27:       index = index + 1;
; 7. 28:    }
; 7. 29:    return 0;
; 7. 30: }
hello